File: PAT (Basic Level) Practise/1007. 素数对猜想/countPrime.cpp
#include <iostream>
#include <math.h>
using namespace std;
bool prime(int i){
	for(int j=2; j<=sqrt(i); j+=1){
		if(i%j==0) return false;
	}
	return true;
}

int main(){
	int start, end;
	int index = 1;
	int cur = 2;
	int i = 3;
	cin>>start>>end;
	if(start == 1){
		if(end>start)
			cout<<"2 ";
		else cout<<"2";
		index++;
	}

	while(cur <= end){
		if(prime(i)){
			if(cur == end){
				cout<<i;
				index++;
			}
			else if(cur >= start){
				if(index%10==0)
					cout<<i<<endl;
				else cout<<i<<" ";
				index++;
			}
			cur++;
		}
		i+=2;
	}
	return 0;
}

File: PAT (Basic Level) Practise/1013. 数素数/CountPrime.py
__author__ = 'Just_CJ'

import math
import re

def prime(i):
    for j in range(2, int(math.sqrt(i))+1):
        if i%j == 0: return False
    return True

if __name__ == '__main__':

    start, end = map(int, re.split(r'\s+(?!$)', raw_input()))
    cur = 2
    i = 3
    prime_list = []
    if start == 1:
        prime_list.append(2)
    while cur <= end:
        if prime(i):
            if cur >= start:
                prime_list.append(i)
            cur += 1
        i += 2

    for i in range(len(prime_list)):
        if (i+1)%10 == 0:
            print prime_list[i]
        else:
            print prime_list[i],